File: exam_admin_front/src/types/model/dto/RandomSelectQuDTO.ts
// 随机抽取题目DTO
export interface RandomSelectQuDTO {
  excludes: string[];
  repoId: string;
  levels: [
    { level: '0'; quCount: ''; num: 0 },
    { level: '1'; quCount: ''; num: 0 }
  ];
  quType: number;
}

interface ClassfiySelect {
  /**
   * 题目难度
   */
  level?: number;
  /**
   * 题目数量
   */
  quCount?: number;
  /**
   * 选取的题目数量
   */
  num?: number;
}

File: exam_common/src/main/java/com/jl/project/entity/dto/RandomSelectQuDTO.java
package com.jl.project.entity.dto;

import com.jl.project.entity.po.ClassfiySelect;
import io.swagger.v3.oas.annotations.media.Schema;

import java.util.List;

/**
 * 随机抽取题目DTO
 */
@Schema(description = "随机抽取题目DTO")
public class RandomSelectQuDTO {

    /**
     * 题目排除列表
     */
    @Schema(description = "题目排除列表")
    private List<String> excludes;

    /**
     * 题库ID
     */
    @Schema(description = "题库ID")
    private String repoId;

    /**
     * 分类列表
     */
    @Schema(description = "分类列表")
    private List<ClassfiySelect> levels;
    /**
     * 题型
     */
    @Schema(description = "题型")
    private Integer quType;

    public String getRepoId() {
        return repoId;
    }

    public void setRepoId(String repoId) {
        this.repoId = repoId;
    }

    public List<ClassfiySelect> getLevels() {
        return levels;
    }

    public void setLevels(List<ClassfiySelect> levels) {
        this.levels = levels;
    }

    public Integer getQuType() {
        return quType;
    }

    public void setQuType(Integer quType) {
        this.quType = quType;
    }

    public List<String> getExcludes() {
        return excludes;
    }

    public void setExcludes(List<String> excludes) {
        this.excludes = excludes;
    }
}


